web/src/features/charts/graphUtils.ts

```diff
@@ -198,6 +198,14 @@ export function getBadgeText(chartData: AreaGraphElement[], t: TFunction) {
     (day) => day.meta.estimationMethod || day.meta.estimatedPercentage === 100
   );
 
+  const allTimeSlicerAverageMethod = chartData.every(
+    (day) => day.meta.estimationMethod === EstimationMethods.TSA
+  );
```

Comment:
I'm wondering if it doesn't start to make sense to create a custom function for these instead so we only loop over it once and set different Boolean flags instead?

Or at the very least re arrange these so the TSA is checked first and then the early TSA return then the rest of the function as it were.